backend/dataall/modules/notifications/services/notification_service.py

```diff
@@ -0,0 +1,70 @@
+"""
+A service layer for Notifications
+"""
+
+import logging
+from dataall.base.db import exceptions
+
+from dataall.base.context import get_context
+from dataall.modules.notifications.db import notification_models as models
+from functools import wraps
+
+from dataall.modules.notifications.db.notification_repositories import NotificationRepository
+
+logger = logging.getLogger(__name__)
+
+
+def _session():
+    return get_context().db_engine.scoped_session()
+
+
+class NotificationAccess:
+    @staticmethod
+    def is_recipient(f):
+        @wraps(f)
+        def wrapper(*args, **kwds):
+            uri = kwds.get('notificationUri')
+            if not uri:
+                raise KeyError(f"{f.__name__} doesn't have parameter uri.")
+            context = get_context()
+            with context.db_engine.scoped_session() as session:
+                notification = session.query(models.Notification).get(uri)
+                if notification and (notification.recipient in context.groups + [context.username]):
+                    return f(*args, **kwds)
+                else:
+                    raise exceptions.UnauthorizedOperation(
+                        action='UPDATE NOTIFICATION',
+                        message=f'User {context.username} is not the recipient user/group of the notification {uri}',
+                    )
```

Comment:
Would you mind moving the check logic in a standalone static method so that we are able to mock/spy it from the tests? See [here](https://github.com/data-dot-all/dataall/blob/472060afd57fba80300a71ba0f771adbcd621df4/backend/dataall/modules/metadata_forms/services/metadata_form_access_service.py#L12) `can_perform` is the decorator and `is_owner` the implementation.

The reason we cannot mock it like that is that decorators are applied at module load time and we patch much later.
So to workaround it we would have to patch before we import the modules which works but is very strange.